src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupBasedAlleles.java

```diff
@@ -1,25 +1,48 @@
 
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.util.SequenceUtil;
 import htsjdk.variant.variantcontext.Allele;
 import htsjdk.variant.variantcontext.VariantContext;
 import htsjdk.variant.variantcontext.VariantContextBuilder;
 import org.broadinstitute.hellbender.engine.AlignmentAndReferenceContext;
 import org.broadinstitute.hellbender.engine.AlignmentContext;
 import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.PileupDetectionArgumentCollection;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
 
 import java.util.*;
 
 
+/**
+ * Helper class for handling pileup allele detection supplement for assembly. This code is analogous but not exactly
+ * equivalent to the DRAGEN ColumnwiseDetection approach.
+ */
 public final class PileupBasedAlleles {
 
-
-//    ArrayList<VariantContext> forcedPileupAlleles;
-
-
-    public static ArrayList<VariantContext> getPileupVariantContexts(List<AlignmentAndReferenceContext> alignmentAndReferenceContextList) {
+    /**
+     * Accepts the raw per-base pileups stored from the active region detection code and parses them for potential variants
+     * that are visible in the pileups but might be dropped from assembly for any number of reasons. The basic algorithm works
+     * as follows:
+     *  - iterate over every pileup and count alt bases
+     *      - (unfinished) detect insertions overlapping this site (CURRENTLY ONLY WORKS FOR INSERTIONS)
+     *  - count "bad" reads as defined by Illumina filtering for puleup detection of variants {@Link #evaluateBadRead}
+     *  - For each detected alt, evaluate if the number of alternate bases are sufficient to make the call and make a VariantContext.
+     *
+     * @param alignmentAndReferenceContextList  List of stored pileups and reference context information where every element is a base from the active region.
+     *                                          NOTE: the expectation is that the stored pileups are based off of the ORIGINAL (un-clipped) reads from active region determination.
+     * @param args                              Configuration arguments to use for filtering/annotations
+     * @param headerForReads                    Header for the reads (only necessary for SAM file conversion)
+     * @return A list of variant context objects corresponding to potential variants that pass our heuristics.
+     */
+    public static ArrayList<VariantContext> getPileupVariantContexts(final List<AlignmentAndReferenceContext> alignmentAndReferenceContextList, final PileupDetectionArgumentCollection args, final SAMFileHeader headerForReads) {
 
         final ArrayList<VariantContext> pileupSNPsList = new ArrayList<>();
 
+        // Iterate over every base
```

Comment:
in the line above, why name the variable pileupSNPsList if it ends up containing both SNPs and Indels? perhaps rename to pileupVariantList?